src/piv.rs

```diff
@@ -295,6 +311,77 @@ impl From<RetiredSlotId> for u8 {
     }
 }
 
+impl Display for RetiredSlotId {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}",
```

Comment:
This can leverage the `Debug` impl too:

```suggestion
        write!(f, "{:?}", self)
```